web/client/utils/LayersUtils.js

```diff
@@ -71,10 +71,64 @@ const isSupportedLayer = (layer, maptype) => {
     return Layers.isSupported(layer.type) && !layer.invalid;
 };
 
+
 const checkInvalidParam = (layer) => {
     return layer && layer.invalid ? assign({}, layer, {invalid: false}) : layer;
 };
 
+const getNode = (nodes, id) => {
+    if (nodes && isArray(nodes)) {
+        return nodes.reduce((previous, node) => {
+            if (previous) {
+                return previous;
+            }
+            if (node && (node.name === id || node.id === id)) {
+                return node;
+            }
+            if (node && node.nodes && node.nodes.length > 0) {
+                return getNode(node.nodes, id);
+            }
+            return previous;
+        }, null);
+    }
+    return null;
+};
+
+const getGroupNodes = (node) => {
+    if (node && node.nodes) {
+        return node.nodes.reduce((a, b) => {
+            let nodes = [].concat(a);
+            if (b.nodes) {
+                nodes = a.concat(getGroupNodes(b));
+            }
+            if (isString(b)) {
+                nodes.push(b);
```

Comment:
possible side effect. Avoid side effect as much as you can. This function returns the new nodes, so it doesn't have to mutate the existing value.